linux/power_operation_scripts/set_power_script.yml

```diff
@@ -78,33 +82,59 @@
 - name: "Update inventory"
   include_tasks: ../../common/update_inventory.yml
 
-- name: "Check the script file running as expected"
-  ansible.builtin.stat:
-    path: "{{ script_tag_path }}"
+- name: "Get content of file tools.conf after {{ power_script_type }} {{ power_script_op }}"
+  ansible.builtin.command: "cat {{ vmtools_config_file }}"
+  register: tools_conf_content_result
   delegate_to: "{{ vm_guest_ip }}"
-  register: tag_path_exist_result
-  failed_when: tag_path_exist_result.stat.exists is undefined or not tag_path_exist_result.stat.exists
+
+- name: "Check the script file running as expected"
   when:
     - script_tag_path is defined and script_tag_path
     - power_script_type == "custom"
     - power_script_op != "suspend"
+  block:
+    - name: "Get the status of file {{ script_tag_path }} for {{ power_script_type }} {{ power_script_op }}"
+      ansible.builtin.stat:
+        path: "{{ script_tag_path }}"
+      delegate_to: "{{ vm_guest_ip }}"
+      register: tag_path_exist_result
+      until:
+        - tag_path_exist_result.stat.exists is defined
+        - tag_path_exist_result.stat.exists
+      retries: 30
+      delay: 5
+
+    - name: "Check the searching result"
+      ansible.builtin.assert:
+        that:
+          - tag_path_exist_result.stat.exists is defined
+          - tag_path_exist_result.stat.exists
+        fail_msg: "Failed to find {{ script_tag_path }} for {{ power_script_type }} {{ power_script_op }}."
+        success_msg: "Successfully found the {{ script_tag_path }} for {{ power_script_type }} {{ power_script_op }}."
 
 - name: "Check the log file"
   when: power_script_op != "suspend"
   block:
-    - name: "Search the log file"
-      ansible.builtin.shell: "grep '{{ script_file_path }}' /var/log/vmware-tools/*.log"
+    - name: "Search the log file {{ script_file_path }}"
+      ansible.builtin.shell: "grep {{ script_file_path }} *.log"
+      args:
+        chdir: "/var/log/vmware-tools"
       delegate_to: "{{ vm_guest_ip }}"
-      ignore_errors: true
       register: result_search_str
+      until:
+        - result_search_str.rc is defined
+        - result_search_str.rc == 0
+      retries: 30
+      delay: 5
+      ignore_errors: true
 
     - name: "Check the serching result"
       ansible.builtin.assert:
         that:
           - result_search_str.rc is defined
           - result_search_str.rc == 0
-        fail_msg: "Failed to find the {{ script_file_path }} in {{ vmtools_vmsvc_log_file }} for {{ power_script_op }}."
-        success_msg: "Successfully found the {{ script_file_path }} in {{ vmtools_vmsvc_log_file }} for {{ power_script_op }}."
+        fail_msg: "Failed to find the {{ script_file_path }} in {{ vmtools_vmsvc_log_file }} for {{ power_script_type }} {{ power_script_op }}."
+        success_msg: "Successfully found the {{ script_file_path }} in {{ result_search_str.stdout }} for {{ power_script_type }} {{ power_script_op }}."
```

Comment:
I saw an example of the output of this task, and I think it is not friendly for reading the success message. 
```
TASK [Check the serching result] ***********************************************
task path: /home/worker/workspace/Ansible_RHCOS_MAIN-51/ansible-vsphere-gos-validation/linux/power_operation_scripts/set_power_script.yml:131
ok: [localhost] => {
    "changed": false,
    "msg": "Successfully found the /etc/vmware-tools/poweron-vm-default in vmsvc.log:[2024-10-14T05:53:25.854Z] [ message] [powerops] [936] Executing script: '/etc/vmware-tools/poweron-vm-default' for default power."
}
```

Reply:
Done to change it. Thanks